soh/src/code/z_parameter.c

```diff
@@ -1661,24 +1661,46 @@ u8 Item_Give(GlobalContext* globalCtx, u8 item) {
         gSaveContext.inventory.equipment |= (gBitFlags[item - ITEM_BOOTS_KOKIRI] << gEquipShifts[EQUIP_BOOTS]);
         return ITEM_NONE;
     } else if ((item == ITEM_KEY_BOSS) || (item == ITEM_COMPASS) || (item == ITEM_DUNGEON_MAP)) {
-        // if we get a boss/big key in ganon's castle (which doesn't have a map/compass)
-        // when rando'd it's for ganon's tower
-        if (gSaveContext.n64ddFlag && gSaveContext.mapIndex == 13) {
-            gSaveContext.inventory.dungeonItems[10] |= 1;
-        } else {
-            gSaveContext.inventory.dungeonItems[gSaveContext.mapIndex] |= gBitFlags[item - ITEM_KEY_BOSS];
+        // Boss Key, Compass, and Dungeon Map scenemap exceptions for rando.
+        if (gSaveContext.n64ddFlag) {
+            // if we get a boss/big key in ganon's castle (which doesn't have a map/compass)
+            // when rando'd it's for ganon's tower
+            if (gSaveContext.mapIndex == 13) {
+                gSaveContext.inventory.dungeonItems[10] |= 1;
+            } else {
+                gSaveContext.inventory.dungeonItems[gSaveContext.mapIndex] |= gBitFlags[item - ITEM_KEY_BOSS];
+            }
+            // if we get a boss/big key, dungeon map, or compass in Desert Colossus, it is
+            // for Spirit Temple.
+            if (gSaveContext.mapIndex == 11) {
```

Comment:
having it in this order means that you'll set the flag for mapindex 11 and mapindex 6

you probably want
```
if (gSaveContext.mapIndex == 13) {
 [...]
} else if (gSaveContext.mapIndex == 11) {
  [...]
} else {
 [...]
}
```

Reply:
Good catch, fixed in 1fe85c9